### ISSUE TITLE
New line length is tokenized as 0

### ISSUE THREAD

**Issue by aik099** [squizlabs/PHP_CodeSniffer#539]
The `length` property of each token is cached `strlen($tokens[$index]['content'])`. Unfortunately for code below the line endings are have `length` as 0 instead of 1:
```php
<?php
$a = array(5, 'a', 10);
for ($i = 0; $i < 5; $i++) {
}
$a = array(
	5,
);
func('a', 5);
func($a[5], 'b');
```

**Comment by gsherwood** [comment]
This is intentional. Only visible characters are included in the token lengths so you don't have to keep subtracting 1 to determine line length.

Is there a problem you are encountering where you need to know that every newline is 1 character?

**Comment by aik099** [comment]
I'm writing a sniff, that checks for single whitespace after comma. Since newline is considered as whitespace as well it should allow for comma, that is last symbol on a line to be absolutely valid. Right now the encountered whitespace (newline) has 0 length and during auto-fixing it's replaced with 1 space.

For now I'm checking, that `$tokens[($stackPtr + 1)]['content'] === $phpcsFile->eolChar` to detect line ending because technically new line token is on same line as content before it.

**Comment by gsherwood** [comment]
Checking against the `$phpcsFile->eolChar` member var is the way I also do it. I don't think there is anything wrong with that. Glad it is working for you.
